features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java

```
@@ -0,0 +1,151 @@
+            pollStatus.setDeviceConfig(config);
+            return pollStatus;
+        } catch (TimeoutException | InterruptedException | ExecutionException e) {
+            LOG.error("Config retrieval failed", e);
+            return PollStatus.unresponsive("Config retrieval failed : " + e.getMessage());
+        }
+    }
+
+    static private class TftpReceiver implements TftpFileReceiver {
+
+        private final CompletableFuture<byte[]> configFuture = new CompletableFuture<>();
+        private final InetAddress serviceAddress;
+
+        public TftpReceiver(InetAddress serviceAddress) {
+            this.serviceAddress = serviceAddress;
+        }
+
+        @Override
+        public void onFileReceived(InetAddress address, String fileName, byte[] content) {
+            if (serviceAddress.equals(address)) {
+                configFuture.complete(content);
+            }
+        }
+
+        public CompletableFuture<byte[]> getConfigFuture() {
+            return configFuture;
+        }
+    }
+
+    public void setTftpServer(TftpServer tftpServer) {
+        this.tftpServer = tftpServer;
+    }
+
+    public void setSshScriptingService(SshScriptingService sshScriptingService) {
+        this.sshScriptingService = sshScriptingService;
+    }
+
+    private String getObjectAsStringFromParams(Map<String, Object> params, String key) {
+        Object obj = params.get(key);
+        if (obj instanceof String) {
+            return (String) obj;
+        }
+        return null;
+    }
+
+    private Integer getObjectAsIntFromParams(Map<String, Object> params, String key) {
+        Object obj = params.get(key);
+        if (obj instanceof Integer) {
+            return (Integer) obj;
+        }
+        return null;
+    }
+
+    private Long getObjectAsLongFromParams(Map<String, Object> params, String key) {
+        Object obj = params.get(key);
+        if (obj instanceof Long) {
+            return (Long) obj;
+        }
+        return null;
```

Are different types supported in the monitor configuration? Maybe throw an exception if a non-long value was configured.

---

yeah, different types are supported in monitor.   For optional values, i return null since there is fallback default value.

---

I think that `IllegalArgumentException` should be thrown in case the configured parameter is not a long. (A silently ignored value is difficult to diagnose.)

---

Handled